src/constraints/constraints.jl

```diff
@@ -0,0 +1,232 @@
+export
+    DistanceConstraint,
+    check_position_constraints,
+    check_velocity_constraints,
+    disable_intra_constraint_interactions!,
+    apply_position_constraints!,
+    apply_velocity_constraints!    
+
+abstract type ConstraintAlgorithm end;
```

Comment:
Remove this abstract type, unless it is specifically required somewhere.

Reply:
It's used by the `check_position_constraint` (and velo) functions. It's not explicitly needed right now since we only have one algorithm but I don't think its hurting anything?

Reply:
If it's used for type annotation of function arguments then it forces anything calling that function to be a sub-type. If those type annotations are removed (as I am suggesting) then it doesn't seem to be used for anything.